Apps/WebClient/src/ClientApp/src/views/credentials.vue

```diff
@@ -0,0 +1,110 @@
+<script lang="ts">
+import Vue from "vue";
+import { Component } from "vue-property-decorator";
+import { Action, Getter } from "vuex-class";
+
+import CredentialInstructionsView from "@/components/credential/credentialInstructions.vue";
+import CredentialManagementView from "@/components/credential/credentialManagement.vue";
+import LoadingComponent from "@/components/loading.vue";
+import ResourceCentreComponent from "@/components/resourceCentre.vue";
+import { WalletConnection } from "@/models/credential";
+import { ImmunizationEvent } from "@/models/immunizationModel";
+import User from "@/models/user";
+import { SERVICE_IDENTIFIER } from "@/plugins/inversify";
+import container from "@/plugins/inversify.container";
+import { ILogger } from "@/services/interfaces";
+
+@Component({
+    components: {
+        LoadingComponent,
+        "credential-instructions": CredentialInstructionsView,
+        "credential-management": CredentialManagementView,
+        "resource-centre": ResourceCentreComponent,
+    },
+})
+export default class CredentialsView extends Vue {
+    @Getter("user", { namespace: "user" })
+    private user!: User;
+
+    @Getter("covidImmunizations", { namespace: "immunization" })
+    covidImmunizations!: ImmunizationEvent[];
+
+    @Getter("connection", { namespace: "credential" })
+    connection!: WalletConnection | undefined;
+
+    @Getter("isLoading", { namespace: "user" })
+    isPatientLoading!: boolean;
+
+    @Getter("isLoading", { namespace: "immunization" })
+    isImmunizationLoading!: boolean;
+
+    @Action("getPatientData", { namespace: "user" })
+    getPatientData!: () => Promise<void>;
+
+    @Action("retrieve", { namespace: "immunization" })
+    retrieveImmunizations!: (params: { hdid: string }) => Promise<void>;
+
+    @Action("retrieveConnection", { namespace: "credential" })
+    retrieveConnection!: (params: { hdid: string }) => Promise<boolean>;
+
+    @Action("retrieveCredentials", { namespace: "credential" })
```

Comment:
i think we might have to get rid of this, action and use the connection to retrieve the list of credentials

Reply:
We cold use this if we want to refresh the credentials without having to retrieve the connection as well

Reply:
Can discuss changing it later - right now it's out of scope of this task.